### PR TITLE
Channelz part 10: GetServer

### PR DESCRIPTION
**Pull request by ncteisen**
Tracking #15340 

Built on #16990

Proposed in grpc/proposal#114

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                 FILE SIZE
 ++++++++++++++ GROWING                                   ++++++++++++++
  +0.0%    +120 [None]                                       +592  +0.0%
  +3.7%    +128 src/core/lib/channel/channelz_registry.cc    +128  +3.7%
      [NEW]    +117 grpc_channelz_get_server                     +117  [NEW]
      +7.4%     +11 [Unmapped]                                    +11  +7.4%

  +0.0%    +248 TOTAL                                        +720  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +16  +0.0%

  [ = ]       0 TOTAL      +16  +0.0%



```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,014,439      Total (=)      2,014,439

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,113,822      Total (>)     11,113,671

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                 FILE SIZE
 ++++++++++++++ GROWING                                   ++++++++++++++
  +0.0%    +120 [None]                                       +592  +0.0%
  +3.7%    +128 src/core/lib/channel/channelz_registry.cc    +128  +3.7%
      [NEW]    +117 grpc_channelz_get_server                     +117  [NEW]
      +7.4%     +11 [Unmapped]                                    +11  +7.4%

  +0.0%    +248 TOTAL                                        +720  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +16  +0.0%

  [ = ]       0 TOTAL      +16  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,014,439      Total (=)      2,014,439

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,113,824      Total (>)     11,113,673

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

File: include/grpc/grpc.h [hunk 1]
```diff
@@ -506,6 +506,9 @@ GRPCAPI char* grpc_channelz_get_top_channels(intptr_t start_channel_id);
 /* Gets all servers that exist in the process. */
 GRPCAPI char* grpc_channelz_get_servers(intptr_t start_server_id);
 
+/* Returns a single Server, or else a NOT_FOUND code. */
+GRPCAPI char* grpc_channelz_get_server(intptr_t server_id);
```
Comment (markdroth) [inline include/grpc/grpc.h]: I assume we're also going to need to add this method to the C++ service implementation?

Comment (ncteisen) [reply include/grpc/grpc.h]: Yep, I was going to do that in a follow up with the appropriate testing in the channelz_service_test file

**Pull request merged**